azurelinuxagent/common/cgroupapi.py

```diff
@@ -327,20 +340,26 @@ def pre_exec_function():
                     try:
                         self._add_process_to_cgroup(pid, cgroup.path)
                     except Exception as exception:
-                        logger.warn("Failed to add PID {0} to the cgroups for extension '{1}'. Resource usage will not be tracked. Error: {2}".format(pid, extension_name, ustr(exception)))
+                        logger.warn("Failed to add PID {0} to the cgroups for extension '{1}'. "
+                                    "Resource usage will not be tracked. Error: {2}".format(pid,
+                                                                                            extension_name,
+                                                                                            ustr(exception)))
             except Exception as e:
-                logger.warn("Failed to add extension {0} to its cgroup. Resource usage will not be tracked. Error: {1}".format(extension_name, ustr(e)))
+                logger.warn("Failed to add extension {0} to its cgroup. Resource usage will not be tracked. "
+                            "Error: {1}".format(extension_name, ustr(e)))
 
-        process = subprocess.Popen(
-            command,
-            shell=shell,
-            cwd=cwd,
-            stdout=stdout,
-            stderr=stderr,
-            env=env,
-            preexec_fn=pre_exec_function)
+        self.track_cgroups(extension_cgroups)
```

Comment:
Can you double-check the code in the telemetry thread? I believe it'll drop the cgroup if it does not have any processes. If that is the case, then we have a raise condition here, because we are starting the process after requesting the telemetry thread to track them

Reply:
We do a cleanup of inactive cgroups every time we poll them to report their telemetry, which is every 5 minutes. It is a race condition, I'll refactor the code to start tracking immediately after invoking the subprocess.